api/types/integration.go

```diff
@@ -0,0 +1,266 @@
+			Kind:     KindIntegration,
+			Version:  V1,
+			SubKind:  IntegrationSubKindAWSOIDC,
+		},
+		Spec: IntegrationSpecV1{
+			SubKindSpec: &IntegrationSpecV1_AWSOIDC{
+				AWSOIDC: spec,
+			},
+			Status: IntegrationSpecV1_INTEGRATION_STATUS_PAUSED,
+		},
+	}
+	if err := ig.CheckAndSetDefaults(); err != nil {
+		return nil, trace.Wrap(err)
+	}
+	return ig, nil
+}
+
+// String returns the integration string representation.
+func (ig *IntegrationV1) String() string {
+	return fmt.Sprintf("IntegrationV1(Name=%v, SubKind=%s, Status=%s, Labels=%v)",
+		ig.GetName(), ig.GetSubKind(), ig.GetStatus(), ig.GetAllLabels())
+}
+
+// MatchSearch goes through select field values and tries to
+// match against the list of search values.
+func (ig *IntegrationV1) MatchSearch(values []string) bool {
+	fieldVals := append(utils.MapToStrings(ig.GetAllLabels()), ig.GetName(), ig.GetSubKind())
+	return MatchSearch(fieldVals, values, nil)
+}
+
+// setStaticFields sets static resource header and metadata fields.
+func (ig *IntegrationV1) setStaticFields() {
+	ig.Kind = KindIntegration
+	ig.Version = V1
+}
+
+// CheckAndSetDefaults checks and sets default values
+func (ig *IntegrationV1) CheckAndSetDefaults() error {
+	ig.setStaticFields()
+	if err := ig.ResourceHeader.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	return trace.Wrap(ig.Spec.CheckAndSetDefaults())
+}
+
+// CheckAndSetDefaults validates and sets default values for a integration.
+func (s *IntegrationSpecV1) CheckAndSetDefaults() error {
+	if s.SubKindSpec == nil {
+		return trace.BadParameter("missing required subkind spec")
+	}
+
+	switch integrationSubKind := s.SubKindSpec.(type) {
+	case *IntegrationSpecV1_AWSOIDC:
+		err := integrationSubKind.CheckAndSetDefaults()
+		if err != nil {
+			return trace.Wrap(err)
+		}
+	default:
+		return trace.BadParameter("unknown integration subkind: %T", integrationSubKind)
+	}
+
+	return nil
+}
+
+// CheckAndSetDefaults validates an agent mesh integration.
+func (s *IntegrationSpecV1_AWSOIDC) CheckAndSetDefaults() error {
+	if s == nil || s.AWSOIDC == nil {
+		return trace.BadParameter("aws_oidc is required for %q subkind", IntegrationSubKindAWSOIDC)
+	}
+
+	if s.AWSOIDC.RoleARN == "" {
+		return trace.BadParameter("role_arn is required for %q subkind", IntegrationSubKindAWSOIDC)
+	}
+
+	return nil
+}
+
+// GetAWSOIDCIntegrationSpec returns the specific spec fields for `aws-oidc` subkind integrations.
+func (ig *IntegrationV1) GetAWSOIDCIntegrationSpec() *AWSOIDCIntegrationSpecV1 {
+	return ig.Spec.GetAWSOIDC()
+}
+
+// SetAWSOIDCIntegrationSpec sets the specific fields for the `aws-oidc` subkind integration.
+func (ig *IntegrationV1) SetAWSOIDCIntegrationSpec(awsOIDCSpec *AWSOIDCIntegrationSpecV1) {
+	ig.Spec.SubKindSpec = &IntegrationSpecV1_AWSOIDC{
+		AWSOIDC: awsOIDCSpec,
+	}
+}
+
+// GetStatusCode returns the integration status as a code.
+// It can be one of:
+// - paused: integration was just configured or was disabled by the user
```

Comment:
What's the use-case for paused vs running vs error state?

Reply:
I'll get back to this comment after resolving the comment in the RFD PR
https://github.com/gravitational/teleport/pull/22556#discussion_r1157561311

Reply:
Should this now be removed?

Reply:
Yes, just remove all the Integration.Status reference :+1: